### PR TITLE
Add sklearn2json converter

### PR DESCRIPTION
**Pull request by demarley**
Converts a simple MLPClassifier from scikit-learn to the proper JSON format.  
Some of the structure is hard-coded because I don't think the MLPClassifier has the same flexibility as Keras, but this could be wrong.

### CODE REVIEW

**Comment by dguest** [comment]
Hi @demarley, 

First off, thanks for this! We welcome any new converters even if they are a bit rough around the edges. Have you verified that this is doing what you think it should be doing (i.e. that the output is the same as you get from sklearn on a simple test pattern)? If so I'll merge it.

**Comment by demarley** [comment]
Hey @dguest,
It seems the output is _slightly_ different than the predicted values from scikit-learn.  
I'm following up to make sure all of my inputs are correct and such.  
Please hold off on merging until I confirm it is working as expected.

**Comment by demarley** [comment]
Validated that the output from scikit-learn and lwtnn are the same.  Please merge when you are ready.

**Comment by dguest** [comment]
OK, thanks for the contribution!

I'll merge this now, but I should note that with the keras converters we always managed to get agreement between keras and lwtnn out to the precision of the cout statements. If you're seeing something with less precise agreement, it's possible that something isn't defined the same way between sklearn and keras, or that something else is wrong.

**Comment by demarley** [comment]
Hey @dguest, 
From what I can tell the disagreement between python/lwtnn seems to be due to float/double precision (the ntuple values are stored as floats, and lwtnn uses doubles) and at more significant digits than anyone would "reasonably" use (...perhaps).  
I don't see any cause for concern, but maybe there's something I've missed?  
Here's an example:

`lwtnn`: 
`0.0270938 | 0.22055  | 0.748874 | 0.00301339 | 0.000468875`

`scikit-learn`:
`0.027090  | 0.220560 | 0.748860 | 0.003013    | 0.000469`

**Comment by jwsmithers** [comment]
Hi Dan, 
I remember seeing something like this for the BN layer and it came down to me not implementing it exactly right. (The epsilon parameter wasn’t implemented correctly or something) Afterwards I was able to get exact matching up to like 5 or 6 decimal places.
The first number here is the problem which is a bit harder to justify with rounding/precision I think.
But I also could be wrong... :)

**Comment by demarley** [comment]
Hey @jwsmithers ,
Thanks for the info!  
I will look into how the layers are implemented more closely and see if I can find some discrepancies.  
By "first number" are you referring to the `0.0270938` vs `0.027090`?  These agree to 5 decimal places, which should be okay, I hope?

**Comment by dguest** [comment]
Yeah, I agree with josh here: this would register as "wrong" in most of our tests.

For comparison, we typically set `warn_threshold=0.000001` meaning yours would be firing some warnings in the CI tests (which we don't see). So what you have might be fine for most applications, but the disagreement isn't something we normally see and thus I'd be suspicious.

**Comment by demarley** [comment]
Thanks @dguest for the clarification.  
I'll begin looking into issues with the converter and post a new PR when I find a fix.

**Comment by dguest** [comment]
It's also possible that you're just seeing some round off error we've never seen before, so take what I say with a grain of salt. How deep is your network, and what kind of layers does it use?

**Comment by demarley** [comment]
The network has 59 inputs, three hidden layers with 40 nodes (`relu` activations) and a 5-d output using `softmax`.

**Comment by dguest** [comment]
Weird. That doesn't seem terribly abnormal. I wonder if you also get a discrepancy if you use e.g. a dummy input with no hidden layers, 5 inputs, and 5 softmax outputs. I guess I could believe that round off error could accumulate over that network, but if that's what's happening I'd expect better agreement from a very simple one.

**Comment by demarley** [comment]
This is certainly something I will check to look for any errors!
